### PR TITLE
Sscs 5030 appeal to proceed

### PR DESCRIPTION
**Pull request by srochani**
**Before creating a pull request make sure that:**
Added InterlocDocument and struckout document model to deserialise json to model
 
- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/SscsInterlocDecisionDocument.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Builder;
+import lombok.Value;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@Value
+@Builder(toBuilder = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class SscsInterlocDecisionDocument {
```
Comment (arcotc) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/SscsInterlocDecisionDocument.java]: The 3 document class definitions all look the same.  Why not just have one SscsDocument class?

File: src/test/java/uk/gov/hmcts/reform/sscs/ccd/deserialisation/SscsCaseCallbackDeserializerTest.java [hunk 2]
```diff
@@ -108,4 +104,36 @@ public void should_convert_checked_exception_to_runtime_on_error() throws IOExce
...
+        Callback<SscsCaseData> actualSscsCaseCallback = sscsCaseCallbackDeserializer.deserialize(json);
+
+        SscsInterlocDecisionDocument sscsInterlocDecisionDocument = actualSscsCaseCallback.getCaseDetails().getCaseData().getSscsInterlocDecisionDocument();
+
+        assertEquals("2019-06-26", sscsInterlocDecisionDocument.getDocumentDateAdded());
+        assertEquals("DecisionNotice.pdf", sscsInterlocDecisionDocument.getDocumentFileName());
+        assertEquals("Decision Notice", sscsInterlocDecisionDocument.getDocumentType());
+    }
+
+    @Test
```
Comment (arcotc) [inline src/test/java/uk/gov/hmcts/reform/sscs/ccd/deserialisation/SscsCaseCallbackDeserializerTest.java]: What about a test for the SscsInterlockDirectionDocument type?

Comment (srochani) [reply src/test/java/uk/gov/hmcts/reform/sscs/ccd/deserialisation/SscsCaseCallbackDeserializerTest.java]: I will add it now

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/SscsInterlocDecisionDocument.java (hunk 1 above)
Comment (srochani) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/SscsInterlocDecisionDocument.java]: We are using implicit parser which looks for name SscsInterlocDecisionDocument/SscsInterlocDirectionDocument and we don't have much fields in each class

**Pull request merged**